### ISSUE TITLE
bad performance in the same wild video

### ISSUE THREAD

**Issue by bucktoothsir** [facebookresearch/VideoPose3D#6]
![_01b44388-2391-4cd5-94c5-f599b690f4c7](https://user-images.githubusercontent.com/1781061/49889048-81634800-fe7b-11e8-9004-8f76b612aed2.png)
![_69d6714a-4b64-42c0-ab0e-92008275b34e](https://user-images.githubusercontent.com/1781061/49889055-845e3880-fe7b-11e8-9d63-6c68a47edf39.png)

**Comment by Godatplay** [comment]
In terms of the output resolution, you set that with `--viz-size` I chose 10 and it seems close, the default is 5.

I'm not sure how much difference it'll make, but consider also changing center as well since all 3 are used to renormalize the camera.

How did you build your dataset file?

**Comment by dariopavllo** [comment]
Did you follow the instructions mentioned in my last post [here](https://github.com/facebookresearch/VideoPose3D/issues/2#issuecomment-446781517)?

Also, in [this comment](https://github.com/facebookresearch/VideoPose3D/issues/2#issuecomment-443502874) I mentioned that we used CPN to extract the 2D keypoints for the videos in the wild, which produces slightly better results. Anyway, if you followed the steps correctly, Detectron poses should be very similar.

We took the video from YouTube as well, in 1080p resolution.

**Comment by wishvivek** [comment]
@bucktoothsir Regarding getting visualizations of in-the-wild videos, in the second step, where you converted the input video to individual frames, how did you preprocess this incoming frame (scale, crop, center, etc.?) before getting the output from the Detectron?

**Comment by bucktoothsir** [comment]
@Godatplay


> In terms of the output resolution, you set that with `--viz-size`. I chose 10 and it seems close, the default is 5.
> 
> I'm not sure how much difference it'll make, but consider also changing center as well since all 3 are used to renormalize the camera.
> 
> How did you build your dataset file?

your advices works. thanks. Now I get a high resolution output, but the performance remains bad.

I built a dataset file as the same structure as original dataset file. Specifically, I built a fake 3d dataset file and a 2d dataset file. The structure is 'S0/skating' and you could rename subjects and actions, then change the corresponding name in your test scripts.

**Comment by bucktoothsir** [comment]
> @bucktoothsir Regarding getting visualizations of in-the-wild videos, in the second step, where you converted the input video to individual frames, how did you preprocess this incoming frame (scale, crop, center, etc.?) before getting the output from the Detectron?

I didn't take any preprocessing steps.

**Comment by bucktoothsir** [comment]
> In terms of the output resolution, you set that with `--viz-size`. I chose 10 and it seems close, the default is 5.
> 
> I'm not sure how much difference it'll make, but consider also changing center as well since all 3 are used to renormalize the camera.
> 
> How did you build your dataset file?

I also write a dataset file by myself.

**Comment by wishvivek** [comment]
@bucktoothsir Thanks for the response. Also, I'm trying to get keypoints on my images using the Detectron Model (using the R-50-FPN End-to-End Keypoint-Only Mask R-CNN Baseline model, [in this page](https://github.com/facebookresearch/Detectron/blob/master/MODEL_ZOO.md)), using the command:

python Detectron.pytorch/tools/infer_simple.py --dataset coco --cfg Detectron.pytorch/configs/baselines/e2e_keypoint_rcnn_R-50-FPN_1x.yaml --load_detectron Detectron.pytorch/data/pretrained_model/e2e_keypoint_rcnn_R-50-FPN_1x.pkl --image_dir videoframes --output_dir Detectron.pytorch/keypoints

but getting this error:

**Comment by bucktoothsir** [comment]
@wishvivek which version of python do you use?

**Comment by wishvivek** [comment]
@bucktoothsir Thanks, I managed to fix that error. I now have the 3D predictions from the model for my in-the-wild video, but they're all normalized (i.e., [-1,1]). So, I'm wondering how do I unnormalize these 3D predictions? Any help would be great, thanks!

**Comment by lxy5513** [comment]
How to get keypts and bboxes ?

for 12_2017_baselines/e2e_keypoint_rcnn_R-101-FPN_s1x.yaml.
Is there any already traind model to get 2D keyps and bboxes?
like this /path/to/e2e_keypoint_rcnn_R-101-FPN_s1x.kpl

or I need to train on Detetron to get the model?Anyone can help me? thanks a lot

**Comment by bucktoothsir** [comment]
> How to get keypts and bboxes ?
> 
> for 12_2017_baselines/e2e_keypoint_rcnn_R-101-FPN_s1x.yaml.
> Is there any already traind model to get 2D keyps and bboxes?
> like this /path/to/e2e_keypoint_rcnn_R-101-FPN_s1x.kpl
> 
> or I need to train on Detetron to get the model?Anyone can help me? thanks a lot

I used detectron as the author's advice.

**Comment by tobiascz** [comment]
Thanks @bucktoothsir to point me to this issue! 

As I already mentioned in #2 I also was able to run the code on a in the wild example with my own [fork](https://github.com/tobiascz/VideoPose3D) of this repository. I also have some notes for [Detectron](https://github.com/tobiascz/VideoPose3D/tree/master/detectron_tools) in there for the people with difficulties. My 3D results are also way worse than the results created by @dariopavllo. I think my 2D poses are not accurate enough - also thanks to @lxy5513 who also suggested that.

So my next step would be to actually run the detectron poses through CPN to get better 2D results! If someone has another opinion please share maybe I did something wrong in my code?
 
![myOutput](https://github.com/tobiascz/VideoPose3D/blob/master/images/scater_girl.gif?raw=true)

### Authors output
![authorsOutput](https://github.com/facebookresearch/VideoPose3D/blob/master/images/demo_yt.gif?raw=true)

**Comment by YCyuchen** [comment]
@Godatplay @tobiascz I use the inference code to run my own video, taking Detecton's 2d keypoints as input.  the buttocks in my output seems fixed, while i think it should move. Have you met  similar problem? Is there any potential solution i can try to improve the result？
**My output**
![crop-sport-lift2](https://user-images.githubusercontent.com/29044063/74590835-1d26b000-504d-11ea-9d03-f1c2b5dd5e08.gif)

**Comment by tobiascz** [comment]
Hey @YCyuchen,

The reason for that is that the 3D Skeleton is always visualized relative to the center hip joint (you called it buttock). To avoid this you could use the ankles as the relative center of the visualization.
In your test video you can see that while the person is crouching the legs actually go up  in the reconstruction. 
The way this works in the code is that from each joint the x,y and z coordinate of the hip joint is subtracted. You would need to change this and instead subtract from each joint the ankle joint.
